File: 30 Days of Code/Day 6 - Let's Review/Solved.c
/*
Steps for the program:
1. Include necessary header files for input/output and string manipulation.
2. Define the `processString` function to process a string:
    a. Calculate the length of the string.
    b. Print characters at even indices followed by a space.
    c. Print characters at odd indices followed by a newline.
3. In the `main` function:
    a. Read the number of test cases (t).
    b. For each test case:
        i. Read the input string.
        ii. Call `processString` to process and print the string.
*/

#include <stdio.h>
#include <string.h>

void processString(char *str) {
    int len = strlen(str);
    // Print even-indexed characters
    for (int i = 0; i < len; i += 2) {
        printf("%c", str[i]);
    }
    printf(" ");
    // Print odd-indexed characters
    for (int i = 1; i < len; i += 2) {
        printf("%c", str[i]);
    }
    printf("\n");
}

int main() {
    int t;
    scanf("%d", &t);
    char str[10000];
    for (int i = 0; i < t; i++) {
        scanf("%s", str);
        processString(str);
    }
    return 0;
}

File: 30 Days of Code/Day 6 - Lets Review/Solved.py
"""
This script processes multiple strings to separate their characters into even-indexed and odd-indexed groups.
    
Steps:
    1. Define a function `processString(s)` that:
        - Extracts characters at even indices from the input string `s`.
        - Extracts characters at odd indices from the input string `s`.
        - Prints the even-indexed characters and odd-indexed characters separated by a space.
    2. Read an integer `t` from the input, representing the number of test cases.
    3. For each test case:
        - Read a string `s` from the input.
        - Call the `processString(s)` function to process and display the result.
"""

def processString(s):
    even_chars = s[::2]
    odd_chars = s[1::2]
    
    print(f"{even_chars} {odd_chars}")
    
t = int(input())

for _ in range(t):
    s = input().strip()
    processString(s)